examples/ota-requestor-app/esp32/main/BDXDownloader.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "BDXDownloader.h"
```

Comment:
Very simliar to `BDXOtaSender`, we'd like to avoid duplicate code with the common implementation. For the downloader it means we need to separate the downloading function and the write image function.

In fact this file fulfills the abstraction quite well with `mOnBlockReceivedCallback`. Can we use this file as the new common implementation?